### PR TITLE
Platform/kpm

### PR DESCRIPTION
**Pull request by mrkno**

### CODE REVIEW

File: core/modules.js [hunk 1]
```diff
@@ -56,25 +68,21 @@ exports.listModules = function (callback) {
...
+				continue;
+			}
 			
             if (!kj.folderPath) {
                 kj.folderPath = folderPath;
             }
             modules[kj.name] = kj;
         } catch (e) {
-            console.critical(e);
+            //console.critical(e);
```
Comment (Awarua-) [inline core/modules.js]: Commented out?

Comment (mrkno) [inline core/modules.js]: Fixed

File: core/core_modules/kpm.js [hunk 2]
```diff
@@ -1,25 +1,279 @@
+var gitpull = require.safe('git-pull'),
+    gitclone = require.safe('git-clone'),
+    files = require.once('../files.js'),
+    modules = require.once('../modules.js'),
+    cfg = require.once('../config.js'),
+    path = require('path'),
+    fs = require.safe('fs-extra'),
+    rmdir = require.safe('rimraf'),
+    tmp = require('tmp'),
```
Comment (Awarua-) [inline core/core_modules/kpm.js]: require.safe

File: core/core_modules/kpm.js [hunk 3]
```diff
@@ -1,25 +1,279 @@
...
+        gitpull(module.folderPath, function (err, consoleOutput) {
+            if (err) {
+                api.sendMessage('Update failed. Manual intervention is probably required.', event.thread_id);
+            } else {
+                api.sendMessage('Restarting module "' + module.name + '"...');
+                // unload the current version
+                cfg.saveModuleConfig(module.name);
+                this.loadedModules = this.loadedModules.filter(function (value) {
+                    if (value.name === module.name) {
+                        // TODO: save configuration
```
Comment (Awarua-) [inline core/core_modules/kpm.js]: hmm TODO?

File: core/core_modules/kpm.js [hunk 4]
```diff
@@ -1,25 +1,279 @@
...
+                    console.critical(e);
+                    api.sendMessage('Could not install module from "' + url + '".', event.thread_id);
+                    cleanup();
+                }
+            }.bind(this));
+        }.bind(this));
+    };
+
+exports.match = function (text, commandPrefix) {
+    console.log(commandPrefix);
```
Comment (Awarua-) [inline core/core_modules/kpm.js]: Remove

File: core/core_modules/kpm.js [hunk 5]
```diff
@@ -1,25 +1,279 @@
...
+    return text.startsWith(commandPrefix + 'kpm');
 };
 
+exports.run = function (api, event) {
+    var commands = event.body.split(' ');
+    var command = null;
+    if (commands.length < 2 || !opts[(command = commands[1].toLowerCase())]) {
```
Comment (Awarua-) [inline core/core_modules/kpm.js]: This is hard logic to follow, consider refactoring

File: core/modules.js [hunk 6]
```diff
@@ -94,6 +101,7 @@ exports.loadModule = function (module) {
                 m = require.once(startPath);
             }
         } catch (e) {
+			console.critical(e);
```
Comment (Awarua-) [inline core/modules.js]: Indentation

Comment (mrkno) [inline core/modules.js]: fixed

File: core/core_modules/kpm.js (hunk 5 above)
Comment (mrkno) [inline core/core_modules/kpm.js]: fixed

File: core/core_modules/kpm.js (hunk 4 above)
Comment (mrkno) [inline core/core_modules/kpm.js]: done

File: core/core_modules/kpm.js (hunk 3 above)
Comment (mrkno) [inline core/core_modules/kpm.js]: removed

**Pull request merged**